api/envoy/extensions/filters/network/postgres_proxy/v3alpha/postgres_proxy.proto

```diff
@@ -29,4 +29,14 @@ message PostgresProxy {
   // are parsed. Parsing is required to produce Postgres proxy filter
   // metadata. Defaults to true.
   google.protobuf.BoolValue enable_sql_parsing = 2;
+
+  // Controls whether to terminate SSL session initiated by a client.
+  // If the value is false, the Postgres proxy filter will not try to
+  // terminate SSL session, but will pass all the packets to the upstream server.
+  // If the value is true, the Postgres proxy filter will try to terminate SSL
+  // session. In order to do that, the filter chain must use :ref:`starttls transport socket
+  // <envoy_api_msg_extensions.transport_sockets.starttls.v3.StartTlsConfig>`.
+  // If the filter does not manage to terminate the SSL session, it will close the connection from the client.
+  // Defaults to false.
+  google.protobuf.BoolValue terminate_ssl = 3;
```

Comment:
Can you elaborate on why we need to do this via StartTls if this is static configuration? Why can't the listener just be configured with TLS support?

Reply:
I see, this is covered in https://github.com/envoyproxy/envoy/issues/10942.

Reply:
I was just writing the reply. Yes, everything is captured in #10942. TLS is negotiated in-band and protocol starts in clear-text.